schema/validator.go

```diff
@@ -44,6 +52,16 @@ func (v Validator) Validate(src io.Reader) error {
 		return errors.Wrap(err, "unable to read the document file")
 	}
 
+	if f, ok := mapValidateDescendants[v]; ok {
+		if f == nil {
+			return fmt.Errorf("internal error: shame on you for putting a nil value in mapValidateDescendents")
```

Comment:
This is taking me too literally ;).  How about:

    return fmt.Errorf("internal error: mapValidateDescendents[%q] is nil", v)